### ISSUE TITLE
Gate blocked on master testmtlswithauthpolicy

### ISSUE THREAD

**Issue by sdake** [istio/istio#15286]
(NOTE: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities/>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>
)

**Bug description**
gate is failing with
```
Flag --export has been deprecated, This flag is deprecated and will be removed in future.
apiVersion: authentication.istio.io/v1alpha1
kind: MeshPolicy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"authentication.istio.io/v1alpha1","kind":"MeshPolicy","metadata":{"annotations":{},"labels":{"app":"security","chart":"security","heritage":"Tiller","release":"istio"},"name":"default","namespace":""},"spec":{"peers":[{"mtls":{"mode":"PERMISSIVE"}}]}}
  generation: 1
  labels:
    app: security
    chart: security
    heritage: Tiller
    release: istio
  name: default
  selfLink: /apis/authentication.istio.io/v1alpha1/meshpolicies/default
spec:
  peers:
  - mtls:
      mode: PERMISSIVE
error: error parsing /tmp/kubedelete569964428.yaml: error converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context
authn_policy_test.go:49: command failed: "error: error parsing /tmp/kubedelete569964428.yaml: error converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context\n" exit status 1
```

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[X] Developer Infrastructure

**Expected behavior**

no gate fialures

**Steps to reproduce the bug**

Just submit a dummy PR and the gate will fail with this signature.  This is a recent regression in master

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
master: d78245f533d6697c27fd1b5510df2fd2f739b507

**How was Istio installed?**
it wasn't

**Environment where bug was observed (cloud vendor, OS, etc)**
linux prow CI

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by sdake** [comment]
Example Failure: https://gubernator.k8s.io/build/istio-prow/pr-logs/pull/istio_istio/15259/istio-pilot-e2e-envoyv2-v1alpha3-master/3119#istioioistiotestse2etestspilot-testmtlswithauthnpolicy

**Comment by sdake** [comment]
This may have been fixed yesterday with commit: 8d263a7e07347c389b834bee82ad4bf8d31864d8

PR: https://github.com/istio/istio/pull/15287 will validate the current gate.

**Comment by sdake** [comment]
The validation PR passed the gate.
